src/correctionlib/highlevel.py

```diff
@@ -180,6 +224,9 @@ def evaluate(
                 for arg in args
                 if not isinstance(arg, (str, int, float))
             ]
+        except NotImplementedError:
+            if any(str(type(arg)).startswith("<class 'dask_awkward.") for arg in args):
```

Comment:
Here also if it is possible to abstractly identify dask delayed collections, it would be nice to handle dask.array as well.
Are we certain that `numpy.asarray(delayed_array)` will never implicitly compute in the future?

Reply:
I think for `dask_awkward` @agoose77 or @douglasdavis made it clear this would not change.

However, for `dask.array`: https://github.com/dask/dask/blob/main/dask/array/core.py#L1699

It will compute on `np.asnumpy`. so we'd have to search by hand. 

We can also check for the `DaskMethodMixin` class members.

Reply:
Do you want to add `dask.array` support or leave that for the future?

Reply:
For the future, I'll add an error in case we get passed one.